[Enhancement] introduce persistent log to snapshot and exportfile during the middle of export

xiaoyong-z opened the pull request:
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：
<!--
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #17005

## Problem Summary(Required) ：
<!-- (Please describe the changes you have made. In which scenarios will this bug be triggered and what measures have you taken to fix the bug?) -->

If fe crash in the middle of export, after restarting fe, fe can not clean the snapshot and exportFile since their path is not persistent in log. In this pr, we fix this problem by introducing additional log to snapshot and exportfile.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
- [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
- [x] 2.5
- [ ] 2.4
- [ ] 2.3
- [ ] 2.2

chaoyli commented on fe/fe-core/src/main/java/com/starrocks/persist/OperationType.java:
```diff
@@ -92,6 +92,7 @@ public class OperationType {
     // load job for only hadoop load
     public static final short OP_EXPORT_CREATE = 36;
     public static final short OP_EXPORT_UPDATE_STATE = 37;
+    public static final short OP_EXPORT_UPDATE_INFO = 38;
```
Adding a new operationType, the roll back need to take the compatibility into consideration.

wanpengfei-git commented:
run starrocks_admit_test

xiaoyong-z committed:
[Enhancement] introduce persistent log to snapshot and exportfile during the middle of export

If fe crash in the middle of export, after restarting fe,
fe can not clean the snapshot and exportFile since their path is not persistent in log.
In this pr, we fix this problem by introducing additional log to snapshot and exportfile.

Signed-off-by: xyz <a997647204@gmail.com>

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 6 / 110 (05.45%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/ExportMgr.java|0|12|00.00%|[381, 383, 384, 385, 386, 387, 389, 390, 391, 392, 394, 396]|
|:large_blue_circle:|com/starrocks/persist/EditLog.java|0|5|00.00%|[427, 428, 429, 1412, 1414]|
|:large_blue_circle:|com/starrocks/journal/JournalEntity.java|0|3|00.00%|[368, 369, 370]|
|:large_blue_circle:|com/starrocks/load/ExportJob.java|6|90|06.67%|[478, 479, 482, 483, 486, 487, 494, 495, 596, 600, 677, 767, 768, 773, 774, 778, 779, 780, 785, 787, 789, 791, 792, 794, 796, 797, 799, 801, 802, 803, 804, 808, 810, 811, 816, 817, 823, 827, 828, 830, 832, 833, 834, 837, 838, 918, 921, 1057, 1058, 1059, 1060, 1061, 1062, 1063, 1064, 1067, 1068, 1069, 1070, 1071, 1072, 1073, 1074, 1077, 1081, 1086, 1087, 1091, 1092, 1093, 1094, 1095, 1096, 1097, 1100, 1102, 1103, 1104, 1105, 1106, 1107, 1108, 1109, 1112]|

wanpengfei-git commented:
run starrocks_admit_test

xiaoyong-z commented:
run starrocks_admit_test
